src/functionaltest/java/com/ericsson/ei/frontend/SubscriptionHandlingFunctionality.java

```diff
@@ -0,0 +1,138 @@
+package com.ericsson.ei.frontend;
+
+import org.junit.*;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+
+import com.ericsson.ei.config.SeleniumConfig;
+import com.ericsson.ei.frontend.pageobjects.IndexPage;
+import com.ericsson.ei.frontend.pageobjects.SubscriptionPage;
+
+public class SubscriptionHandlingFunctionality extends SeleniumBaseClass {
+
+    private static final String DOWNLOADEDTEMPLATEFILEPATH = String.join(File.separator,
+            SeleniumConfig.getTempDownloadDirectory().getPath(), "subscriptionsTemplate.json");
+    private static final String SUBSCRIPTIONTEMPLATEFILEPATH = String.join(File.separator, "src", "functionaltest",
+            "resources", "responses", "SubscriptionTemplate.json");
+    private static final String RELOADTESTFILEPATH = String.join(File.separator, "src", "functionaltest", "resources",
+            "responses", "SubscriptionForUploadCase.json");
+    private static final String SAVETESTFILEPATH = String.join(File.separator, "src", "functionaltest", "resources",
+            "responses", "SubscriptionForSaveCase.json");
+    private static final String UPLOADFILEPATH = String.join(File.separator, "src", "functionaltest", "resources",
+            "responses", "SubscriptionForUploadCase.json");
+
+    @Test
+    public void testSubscription() throws Exception {
+
+        // Open index page
+        IndexPage indexPageObject = new IndexPage(mockedHttpClient, driver, baseUrl);
+        indexPageObject.loadPage();
+
+        // Click on Subscription Handling page button and verify that it is open
+        String headerPath = "//div[@class='container pull-left']//h1";
```

Comment:
I think it's better to use "find by ID" instead of xpath. If a minor thing is changed in the DOM structure this would more easily break than using ID. If an ID is not present i would suggest to add it to the html code :) This applies to all xpaths you are using.

Reply:
Actually, in the current code state, I have option to use either absolute or xpath... so I preffered xpath for the obvious reasons. However, "find by ID" is a better one but as you mentioned html code need to added for ID.